src/types.ts

```diff
@@ -71,6 +71,12 @@ export interface ProviderConfig {
    * Otherwise, all flags will be requested and listened to.
    */
   flags?: LDFlagSet;
+
+  /**
+   * Optionally, the LDClient can be initialised outside of the provider
+   * and passed in, instead of being initialised by the provider.
+   */
```

Comment:
Thanks for doing this. We have to be more vigilant though when passing in the ldClient and only pass the ldClient when the ready event has been emitted. The react sdk does this by default in `initLDClient` [here](https://github.com/launchdarkly/react-client-sdk/blob/6da47ba8c7b270dced1dbecf866b0eb557dee3cc/src/initLDClient.ts#L29). It maybe a good idea to document that here please?